node/src/components/consensus/era_supervisor.rs

```diff
@@ -260,9 +260,24 @@ where
         let our_id = self.public_signing_key;
         let era_rounds_len = params.min_round_len() * params.end_height();
         let min_end_time = start_time + self.highway_config().era_duration.max(era_rounds_len);
-        let should_activate = self.node_start_time < start_time
-            && min_end_time >= timestamp
-            && validators.iter().any(|v| *v.id() == our_id);
+        let mut should_activate = false;
+        if self.node_start_time >= start_time {
+            info!(
+                era = era_id.0,
+                %self.node_start_time, "not voting; node was not started before the era began",
+            );
+        } else if min_end_time < timestamp {
+            info!(
+                era = era_id.0,
+                %min_end_time,
+                "not voting; era started too long ago",
+            );
+        } else if !validators.iter().any(|v| *v.id() == our_id) {
+            info!(era = era_id.0, %our_id, "not voting; not a validator");
+        } else {
+            info!(era = era_id.0, "start voting");
+            should_activate = true;
+        }
```

Comment:
I had mutable variables that are defaults but can be replaced in _some_ of the condition branches but not in others.

Reply:
OK, I'll change it. It's just that that adds three lines saying `false`…

Reply:
You don't have to. I just wanted to share the sentiment. I should probably refrain myself from this kind of comments.

Reply:
Too late! :stuck_out_tongue: